Floofbot/Modules/Administration.cs

```diff
@@ -421,105 +422,50 @@ await foreach(var messageCollection in asyncMessageCollections)
+                if (string.IsNullOrEmpty(user)) // want to view their own warnlog 
                 {
-                    string userID = Regex.Match(user, @"\d{17,18}").Value;
-                    formalWarnings = _floofDB.Warnings.AsQueryable()
-                                                    .Where(u => u.UserId == Convert.ToUInt64(user) && u.GuildId == Context.Guild.Id)
-                                                    .OrderByDescending(x => x.DateAdded).Take(10);
-                    userNotes = _floofDB.UserNotes.AsQueryable()
-                                                    .Where(u => u.UserId == Convert.ToUInt64(user) && u.GuildId == Context.Guild.Id)
-                                                    .OrderByDescending(x => x.DateAdded).Take(10);
+                    embed = await GetWarnings(Context.Message.Author.Id, true, Context.Channel);
+                    if (embed == null)
+                        return;
+                    await Context.Message.Author.SendMessageAsync("", false, embed);
+                    return;
                 }
                 else
                 {
-                    await Context.Channel.SendMessageAsync($"⚠️ Could not find user \"{user}\"");
+                    await Context.Channel.SendMessageAsync("Only moderators can view the warn logs of other users.");
                     return;
                 }
             }
-            else
-            {
-                formalWarnings = _floofDB.Warnings.AsQueryable()
-                    .Where(u => u.UserId == badUser.Id && u.GuildId == Context.Guild.Id)
-                    .OrderByDescending(x => x.DateAdded).Take(10);
-                userNotes = _floofDB.UserNotes.AsQueryable()
-                    .Where(u => u.UserId == badUser.Id && u.GuildId == Context.Guild.Id)
-                    .OrderByDescending(x => x.DateAdded).Take(10);
-            }
-
-            if (formalWarnings.Count() == 0 && userNotes.Count() == 0)
-            {
-                await Context.Channel.SendMessageAsync($"{badUser.Username}#{badUser.Discriminator} is a good noodle. They have no warnings or user notes!");
-                return;
-            }
-
-            EmbedBuilder builder = new EmbedBuilder();
-            builder.Color = ADMIN_COLOR;
-            int warningCount = 0;
-            int userNoteCount = 0;
-            if (badUser == null) // no user, just id in database
-                builder.WithTitle($"Warnings for {user}");
-            else
-                builder.WithTitle($"Warnings for {badUser.Username}#{badUser.Discriminator}");
-            if (formalWarnings.Count() != 0) // they have warnings
+            else // a mod
             {
-                builder.AddField(":warning: | Formal Warnings:", "\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_\\_");
-                foreach (Warning warning in formalWarnings)
+                if (string.IsNullOrEmpty(user)) // want to view their own warnlog 
```

Comment:
To avoid duplicate logic, I'd suggest checking for an empty `user` string first, then in the `else` condition, check if the user is a mod trying to access another user's warnlog.